packages/storybook/.storybook/preview.js

```diff
@@ -108,9 +108,37 @@ export const decorators = [
   ),
 ];
 
+// Sets up a mutation observer to ensure that the storybook docs-root container doesn't get hidden by modals
+const observeDocsRoot = () => {
+  // The target for the observer
+  const targetNode = document.getElementById('docs-root');
+
+  // Configuration options for the observer
+  const config = {attributes: true, childList: false, subtree: false}
+
+  // Callback function that 'resets' the aria-hidden attribute to false
+  const callback = (
+    /** @type {MutationRecord[]} */ mutationList
+  ) => {
+    for (const mutation of mutationList) {
+      if (mutation.attributeName === 'aria-hidden' && targetNode?.getAttribute('aria-hidden') !== 'false') {
+        targetNode?.setAttribute('aria-hidden', 'false');
```

Comment:
This is scoped to only the `#docs-root` element of storybook and is needed because the `hideOthers` function that we call as part of `setupModal` was hiding the docs-root from screen readers, which effectively meant that inside of storybook modals became 'invisible' to screen readers.